### PR TITLE
[Improvement] add springdoc-openapi-ui

### LINKED ISSUE #12331: [Improvement][API] Use springdoc-openapi to replace springfox
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Since springfox will not update.

related doc: https://springdoc.org

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuchanghai**
## Purpose of the pull request
- close issue: #12331 
### relation docs
-  springdoc-openapi-ui doc. and how to config it  https://springdoc.org/#getting-started
- how to change annotation to fit openapi  https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Annotations#quick-annotation-overview

### update annotation
image

result:
image

## Verify this pull request

This pull request is code cleanup without any test coverage.

this close : #12331

### CODE REVIEW

**Comment by kezhenxu94** [comment]
If you are replacing springfox with springdoc, can you remove the springfox dependencies?

**Comment by fuchanghai** [comment]
> If you are replacing springfox with springdoc, can you remove the springfox dependencies?

i'll fix it,thx

**Comment by fuchanghai** [comment]
i have a problem that  can't  autowire two version's OpenAI

**Comment by ruanwenjun** [comment]
> i have a problem that can't autowire two version's OpenAI

Can we use two `GroupedOpenApi ` to represent two versions? I don't go deep into this.

**Comment by fuchanghai** [comment]
> > i have a problem that can't autowire two version's OpenAI
> 
> Can we use two `GroupedOpenApi ` to represent two versions? I don't go deep into this.

Thanks for your reminding. I find what you said seems to be true
image

**Comment by fuchanghai** [comment]
@ruanwenjun  can you help me to fix the problem that can not  pass check dependencies

**Comment by kezhenxu94** [comment]
> @ruanwenjun can you help me to fix the problem that can not pass check dependencies

Please ignore the OWASP Dependency Check failure.

File: dolphinscheduler-dist/release-docs/LICENSE [hunk 1]
```diff
@@ -377,6 +377,7 @@ The text of each license is also included at licenses/LICENSE-[project].txt.
     spring-context-support 5.3.13: https://mvnrepository.com/artifact/org.springframework/spring-context-support/5.3.13, Apache 2.0
     spring-core 5.3.22: https://mvnrepository.com/artifact/org.springframework/spring-core/5.3.22, Apache 2.0
     spring-expression 5.3.13: https://mvnrepository.com/artifact/org.springframework/spring-expression/5.3.13, Apache 2.0
+    springdoc-openapi-ui 1.6.9: https://mvnrepository.com/artifact/org.springdoc/springdoc-openapi-ui/1.6.9, Apache 2.0
```
Comment (ruanwenjun) [inline dolphinscheduler-dist/release-docs/LICENSE:379]: Please remove the license announce your delete and update the version you upgrade.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dist/release-docs/LICENSE (hunk 1 above)
Comment (fuchanghai) [reply dolphinscheduler-dist/release-docs/LICENSE:379]: done

**Comment by fuchanghai** [comment]
@ruanwenjun @kezhenxu94 plz review this PR

**Review by ruanwenjun — APPROVED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java [hunk 2]
```diff
@@ -16,107 +16,47 @@
...
-
-    private ApiInfo apiV2Info() {
-        return new ApiInfoBuilder()
-                .title("Dolphin Scheduler Api Docs")
-                .description("Dolphin Scheduler Api Docs")
-                .version("V2")
+    public GroupedOpenApi publicApi1() {
+        return GroupedOpenApi.builder()
+                .group("v1")
+                .pathsToMatch("/**")
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java]:
```suggestion
               .pathsToExclude("v2/**")
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java [hunk 3]
```diff
@@ -16,107 +16,47 @@
...
-                    return (List<RequestMappingInfoHandlerMapping>) field.get(bean);
-                } catch (IllegalArgumentException | IllegalAccessException e) {
-                    throw new IllegalStateException(e);
-                }
-            }
-        };
+    public GroupedOpenApi publicApi2() {
+        return GroupedOpenApi.builder()
+                .group("v2")
+                .pathsToMatch("/**")
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java]:
```suggestion
                .pathsToMatch("v2/**")
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
It seems you didn't set the right mapping of v1 and v2, and please remove the swagger.properties, this file is used in spring fox.
image

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java (hunk 2 above)
Comment (fuchanghai) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/configuration/OpenAPIConfiguration.java]: thx for you recommend ,but  in my local  ``` /v2/**``` is right. and i had fixed

**Comment by fuchanghai** [comment]
@ruanwenjun plz have a look

**Comment by ruanwenjun** [comment]
@fuchanghai Basically LGTM, please resolved the conflicts.

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**